### ISSUE TITLE
online editing adds CRs

### ISSUE THREAD

**Issue by tillmo** [ontohub/ontohub#1776]
when I edit and existing ontology and just insert a space, a CR gets added at each line. As a consequence, in the diff, each line is marked as changed. See e.g. https://ontohub.org/repositories/essli_2016/fdf32bd4f241662ba7baaa59de12b6c269b16957/diff

**Comment by eugenk** [comment]
To do: Implement what is said in this comment https://github.com/codemirror/CodeMirror/issues/3395#issuecomment-122898922

**Comment by tillmo** [comment]
or better switch to ACE...

**Comment by phyrog** [comment]
@eugenk I'm not sure how to implement this the easy way. We include codemirror as a rails gem, so I can't just apply the patch that is linked there. I could fork `codemirror-rails` though.

**Comment by eugenk** [comment]
Can't you simply add the option in [app/assets/javascripts/codemirror-custom.js.coffee:7-11](https://github.com/ontohub/ontohub/blob/04a3a8e5a6d3f73ac4ad36f7da39e710b5cd17e6/app/assets/javascripts/codemirror-custom.js.coffee#L7-L11)?

**Comment by phyrog** [comment]
Yes I can, but codemirror does not support this option by default and won't do anything with it (unless we use the linked patch).

**Comment by eugenk** [comment]
Then let's try the official repository. I don't even know why we used a non-official repo in the first place.

**Comment by phyrog** [comment]
I created a branch for this, but it's not complete yet. After using `\r\n` as a line separator (it seems `\n` is now the default) for a file that uses `\r\n`, the whole file is printed in one line. I have not yet found out why this happens.
